_drafts/wsl2-as-a-firmware-development-environment.md

```diff
@@ -0,0 +1,485 @@
+  path length limitations have been mitigated by Microsoft, toolchains have not caught
+  up.  I gave up trying to fix it and I'm not sure whether it's a CMake or Ninja
+  problem.  Even after I used a workaround where I shortened the paths by assigning the
+  root build path to a driver letter, still 48 of the 244 tests errored due to
+  some object in the build exceeding the max path length.  This is an open
+  [issue](https://github.com/zephyrproject-rtos/zephyr/issues/36358) at Zephyr.
+- The native Ubuntu run is on a different disk than the Windows and WSL2 tests.
+  While the Windows NVME drive is faster than the USB 3.2 Gen 2 drive, the test is
+  expected to be primarily CPU bound instead of being bound by the 1,000MB/s read
+  and 800MB/s write speeds of the SSD expansion card.
+- Operating systems.  They do stuff.  It's hard to know why and when they do
+  that stuff!  All tests were run with a single Chrome instance open on a
+  default tab and the system set to "maximum performance".  Time permitting, the
+  benchmark would be run many times.
+- Test selection could be refined to cover a wider variety of build scenarios,
+  targets, and toolchains.
+- Couldn't get VMWare to use AMD-V.
+
+### Native Ubuntu 24.04
+
+```
+Operating System: Ubuntu 24.04 LTS                  
+          Kernel: Linux 6.8.0-36-generic
+    Architecture: x86-64
+```
+
+All other platforms are compared to the Ubuntu results because Ubuntu is the
+fastest.
+
+| Metric                  | Result     | Compared to Native Ubuntu |
+| ----------------------- | ---------- | ------------------------- |
+| Duration                | 11m 4.444s | -                         |
+| Average build duration  | 2.723s     | -                         |
+| Workload multiplier     | 1.000      | -                         |
+| Duration of a 40s build | 40.000s    | -                         |
+
+![Screenshot of the native Ubuntu benchmark run]({% img_url zephyr-development-environments/ubuntu-twister-run.png %})
+
+### WSL2 Ubuntu 24.04
+
+```
+  Virtualization: wsl
+Operating System: Ubuntu 24.04 LTS
+          Kernel: Linux 5.15.153.1-microsoft-standard-WSL2
+    Architecture: x86-64
+```
+
+| Metric                  | Result                    | Compared to Native Ubuntu |
+| ----------------------- | ------------------------- | ------------------------- |
+| Duration                | 12m 34.642s               | + 1m 30.642s              |
+| Average build duration  | 3.092s                    | + 0.369s                  |
+| Workload multiplier     | 754.642s / 664.444s       | * 1.136                   |
+| Duration of a 40s build | 40.000s * 1.136 = 45.440s | + 5.440s                  |
+
+![Screenshot of the WSL2 5.15 benchmark run]({% img_url zephyr-development-environments/wsl2-twister-run-2.png %})
+
+I also compiled Kernel 6.1 and 6.6 for WSL2 and saw that Kernel 6.1 shrank the
+performance gap by about 4%.  In my opinion, that's not worth it, but it shows
```

Comment:
Unfortunately I didn't record the 6.6 test.  So the screen shot is indeed for 6.1, which I don't think MS will ever officially release since they're working on 6.6 now.

Reply:
Ah I see thanks for clarifying